src/nvim/CMakeLists.txt

```diff
@@ -120,9 +120,14 @@ if(NOT MSVC)
   set_source_files_properties(
     ${CONV_SOURCES} PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-conversion")
   # gperf generates ANSI-C with incorrect linkage, ignore it.
-  set_source_files_properties(
-    eval.c PROPERTIES COMPILE_FLAGS "${COMPILE_FLAGS} -Wno-static-in-inline -Wno-conversion")
-
+  check_c_compiler_flag(-Wno-static-in-inline HAS_WNO_STATIC_IN_INLINE_FLAG)
```

Comment:
What 4.4 added was not support for a "single optional flag", rather the _general_ realization that unknown `-Wno-` can be safely ignored. If the guard checks against `gcc <= 4.3` then it is obvious when this check later could be removed.  Also if we later need to add some new `-Wno-` which not all compilers have, it's then clear we could add it to the same guard, rather than adding yet another `check_c_compiler_flag` and making this logic even more complex.